boards: Added support for Silicon Labs' BRD4255A Radio board

yonsch opened the pull request:
Added support for the BRD4255A board, as part of efr32_radio.
Support for the EFR32FG13P SoC was added to achieve this.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+			compatible = "silabs,gecko-flash-controller";
+			label = "FLASH_CTRL";
+			reg = <0x400e0000 0x78>;
+			interrupts = <24 0>;
```
```suggestion
			interrupts = <25 0>;
```

chrta replied:
https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/silabs/efr32bg13p.dtsi looks similar to this soc, please also check if the other things (e.g. reg sizes etc) are correct.

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+		usart0: usart@40010000 { /* USART0 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x40010000 0x400>;
+			interrupts = <11 0>, <12 0>;
```
```suggestion
			interrupts = <12 0>, <13 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+		usart1: usart@40010400 { /* USART1 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x40010400 0x400>;
+			interrupts = <19 0>, <20 0>;
```
```suggestion
			interrupts = <20 0>, <21 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+		usart2: usart@40010800 { /* USART2 */
+			compatible = "silabs,gecko-usart";
+			reg = <0x40010800 0x400>;
+			interrupts = <19 0>, <20 0>;
```
```suggestion
			interrupts = <38 0>, <39 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+		leuart0: leuart@4004a000 { /* LEUART0 */
+			compatible = "silabs,gecko-leuart";
+			reg = <0x4004a000 0x400>;
+			interrupts = <21 0>;
```
```suggestion
			interrupts = <22 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x4000c000 0x400>;
+			interrupts = <16 0>;
```
```suggestion
			interrupts = <17 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <0x4000c400 0x400>;
+			interrupts = <16 0>;
```
```suggestion
			interrupts = <40 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+		rtcc0: rtcc@40042000 {
+			compatible = "silabs,gecko-rtcc";
+			reg = <0x40042000 0x184>;
+			interrupts = <29 0>;
```
```suggestion
			interrupts = <31 0>;
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+		gpio: gpio@4000a400 {
+			compatible = "silabs,gecko-gpio";
+			reg = <0x4000a400 0xc00>;
+			interrupts = <9 2 17 2>;
```
```suggestion
			interrupts = <10 2 18 2>;
```

chrta commented on dts/arm/silabs/efr32fg13p233f512gm48.dtsi:
```diff
@@ -0,0 +1,25 @@
+ */
+
+#include <mem.h>
+#include <silabs/efr32fg1p.dtsi>
```
```suggestion
#include <silabs/efr32fg13p.dtsi>
```

chrta commented on dts/arm/silabs/efr32fg13p.dtsi:
```diff
@@ -0,0 +1,206 @@
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+#include "gpio_gecko.h"
+
```
Since https://github.com/zephyrproject-rtos/zephyr/blob/master/dts/arm/silabs/efr32bg13p.dtsi is very similar to this file, it would make sense to extract the common content to a new file efr32xg13p.dtsi and include the new file in this file and from efr32bg13p.dtsi

yonsch commented:
Thanks @chrta , I figured efr32fg13p and efr32bg13p have identical dts files, so I merged them to a single efr32xg13p file.

chrta reviewed (approved):
lgtm

chrta commented:
@yonsch Since the pr in the hal repo is merged now, please update the west.yml to let it point to the current master revision of the hal.

mnkp commented on boards/arm/efr32_radio/doc/index.rst:
```diff
@@ -9,16 +9,19 @@ EFR32 Radio Boards
 - `SLWSTK6000B Mighty Gecko Wireless Starter Kit`_
 - `SLWSTK6061B Proprietary Wireless Starter Kit`_
 - `SLWSTK6006A Mighty Gecko Wireless Starter Kit`_
+- `SLWRB4255A Flex Gecko Wireless Radio Board`_
```
It looks like SiLabs does not provide dedicated starter kit for the SLWRB4255A / BRD4255A radio board. One needs to buy a starter kit for one of the other radio boards or [Wireless Starter Kit Mainboard](https://www.silabs.com/development-tools/wireless/wireless-starter-kit-mainboard).

In any case, this entry is incorrect. We should either remove it or replace by 
- SLWMB4001A Wireless Starter Kit Mainboard

yonsch replied:
Thanks @mnkp , I added " or radio boards" to line 17 to denote that not all the boards on that list come as part of a starter kit, but maybe it should have been clearer. Perhaps, I should keep the original starter kits list, and add another list of supported radio boards, mentioning that the main board should be purchased separately?

mnkp replied:
This list was only meant for starter kits. The new board is already listed under "EFR32 Radio Boards" on line 12.

mnkp reviewed (changes requested):
LGTM, thanks! Apart from a minor issue you should squash all the 3 commits into one.

mnkp commented on boards/arm/efr32_radio/doc/index.rst:
```diff
@@ -20,6 +21,7 @@ Support for EFR32 Radio boards is provided by one of the starter kits
 - `SLWSTK6061B Proprietary Wireless Starter Kit`_
 - `SLWSTK6006A Mighty Gecko Wireless Starter Kit`_
 
+
```
Let's remove this empty line.

mnkp reviewed (commented):
Looks good. Now let's restructure the commits. 

We tend to update `west.yml` in a separate commit. You can check `git log` and see how @chrta has done it. The commit updating `west.yml` should be followed up by a single commit adding BRD4255A radio board.

You can either do it using `git rebase -i` or start from scratch by means of `git reset`. To reuse an existing commit message when redoing the commit you can `git commit --reuse-message=<commit>`. At the end you'll need to force push your updates.

mnkp commented:
Thanks. Sorry for a delay. I missed the notification.

Let's change the description of the commit updating west.yml file. The current one is incorrect. The commit title should reflect its content, not the PR it was added from.

Also, I messed up. This PR adds support for the board but also for the EFR32FG13P SoC. It would be cleaner if we split it in 3 and not 2 commits. e.g.

1)
```
west.yml: Update hal_silabs module

Update west.yml to point to hal_silabs version with support for Silicon
Labs EFR32FG13P SoCs.
```
containing `west.yml`

2)
```
soc: silabs_exx32: Add support for EFR32FG13P SoC

This commit adds support for Silicon Labs EFR32FG13P (Flex Gecko) SoC.
```
containing all files in `dts/` and `soc/` folder + `CODEOWNERS`

3)
```
boards: Add support for Silicon Labs' BRD4255A Radio board

This commit adds support for Silicon Labs BRD4255A (a.k.a. SLWRB4255A)
Flex Gecko Radio Board.
```
containing all files in `boards/` folder

yonsch committed:
west.yml: Update hal_silabs module

Update west.yml to point to hal_silabs version with support for Silicon
Labs EFR32FG13P SoCs.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch committed:
soc: silabs_exx32: Add support for EFR32FG13P SoC

This commit adds support for Silicon Labs EFR32FG13P (Flex Gecko) SoC.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch committed:
boards: Add support for Silicon Labs' BRD4255A Radio board

This commit adds support for Silicon Labs BRD4255A (a.k.a. SLWRB4255A)
Flex Gecko Radio Board.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

yonsch commented:
Thanks @chrta and @mnkp for your time and detailed explanations :)
